pandas/io/pickle.py

```diff
@@ -96,7 +97,7 @@ def to_pickle(
 
     with get_handle(
         filepath_or_buffer,
-        "wb",
+        mode,
```

Comment:
Would it make sense to raise an error if "b" is not in mode?

Reply:
I'd fear that's unreliable. the only real reason for people to pass in a different mode here, is if they want to use one of `tarfile`'s [custom modes](https://docs.python.org/3/library/tarfile.html#tarfile.open) like `w:gz`.

Reply:
as an alternative, we could replace the `mode` argument with an argument that's specific to tarfile?

Reply:
If the user can specify mode as part of the `compression_options`, I would remove mode.

Reply:
removed `mode` in https://github.com/pandas-dev/pandas/pull/44787/commits/9458ecbe116c83e04ff2bd111467caa45a16f650 and added a test ensuring that `compression.mode` can be used to specify compression method in https://github.com/pandas-dev/pandas/pull/44787/commits/d20f31505960515bc43fe0074e52ecd141f1940d